homeassistant/components/light/hue.py

```diff
@@ -149,32 +150,53 @@ def update_lights():
             _LOGGER.exception("Cannot reach the bridge")
             return
 
-        api_states = api.get('lights')
+        api_lights = api.get('lights')
 
-        if not isinstance(api_states, dict):
+        if not isinstance(api_lights, dict):
+            _LOGGER.error("Got unexpected result from Hue API")
+            return
+
+        api_groups = api.get('groups')
+
+        if not isinstance(api_groups, dict):
             _LOGGER.error("Got unexpected result from Hue API")
             return
 
         new_lights = []
+        new_lightgroups = []
 
         api_name = api.get('config').get('name')
         if api_name in ('RaspBee-GW', 'deCONZ-GW'):
             bridge_type = 'deconz'
         else:
             bridge_type = 'hue'
 
-        for light_id, info in api_states.items():
+        for light_id, info in api_lights.items():
             if light_id not in lights:
                 lights[light_id] = HueLight(int(light_id), info,
                                             bridge, update_lights,
                                             bridge_type, allow_unreachable)
                 new_lights.append(lights[light_id])
             else:
                 lights[light_id].info = info
+                lights[light_id].update_ha_state()
+
+        for lightgroup_id, info in api_groups.items():
+            if lightgroup_id not in lightgroups:
+                lightgroups[lightgroup_id] = HueLight(
+                    int(lightgroup_id), info, bridge, update_lights,
+                    bridge_type, allow_unreachable, True)
+                new_lightgroups.append(lightgroups[lightgroup_id])
+            else:
+                lightgroups[lightgroup_id].info = info
+                lightgroups[lightgroup_id].update_ha_state()
 
         if new_lights:
             add_devices(new_lights)
 
+        if new_lightgroups:
```

Comment:
There is no benefit in maintaining `new_lights` and `new_lightgroups`  as two separate lists if you are going to pass both to `add_devices`.